2_run/src/run_glm3.R

```diff
@@ -133,11 +133,17 @@ run_glm3_model <- function(sim_dir, nml_objs, model_config, burn_in, burn_out, e
   tryCatch(
     {
       retry::retry(
-        {glm_time <- system.time({glm_code <- GLM3r::run_glm(sim_lake_dir, verbose = FALSE)})[['elapsed']]
-        output_dates <- glmtools::get_temp(nc_filepath) %>%
-          mutate(date = format(as.Date(lubridate::floor_date(DateTime, 'days')),"%Y-%m-%d")) %>%
-          pull(date)
-        max_output_date <- max(output_dates)
+        {
+          # set max_output_date to NA temporarily, in case netCDF values are NA and 
+          # can't be read. If max_output_date isn't updated with actual max date of 
+          # output below, will trigger the error function
+          max_output_date <- NA
+          glm_time <- system.time({glm_code <- GLM3r::run_glm(sim_lake_dir, verbose = FALSE)})[['elapsed']]
+          output_dates <- glmtools::get_temp(nc_filepath) %>%
+            mutate(date = format(as.Date(lubridate::floor_date(DateTime, 'days')),"%Y-%m-%d")) %>%
+            pull(date)
+          # update max_output_date to max date of output
+          max_output_date <- max(output_dates)
```

Comment:
I am trying to decide if it would be cleaner to do `max_output_date = ifelse(exists("max_output_date"), max_output_date, NA)`  when the tibble is created or keep it like this? Either way gets the job done and using `exists()` might be weird with how `tryCatch()` and `retry()` are working ... I am not sure.

Reply:
Hmm that's a good thought. Just tested it and it does seem to work fine with tryCatch() and retry(). I'd thought it would throw an error when it hit the `until` statement of `retry()`, but it didn't, so I think I'll stick with your suggestion as I agree it is cleaner.